scripts/kconfig/hardenconfig.py

```diff
@@ -53,6 +53,11 @@ def compare_with_hardened_conf(kconf, hardened_kconf_filename):
                     current = None
                 options.append(Option(name=name, current=current,
                                   recommended=recommended, symbol=symbol))
+    for node in kconf.node_iter():
+        for select in node.selects:
+            if (kconf.syms["EXPERIMENTAL"] or kconf.syms["DEPRECATED"]) in select:
```

Comment:
this test is quite wrong :)
It just happens to work for the symbols that select EXPERIMENTAL since `kconf.syms["EXPERIMENTAL"] or kconf.syms["DEPRECATED"]` evaluates to `kconf.syms["EXPERIMENTAL"]`

```suggestion
            if kconf.syms["EXPERIMENTAL"] in select or kconf.syms["DEPRECATED"] in select:
```

Reply:
or maybe a more idiomatic way:

```suggestion
            if any(kconf.syms[key] in select for key in ["EXPERIMENTAL", "DEPRECATED"]):
```

Reply:
ouch, that is right. My head is not in the right place these days :p